### PR TITLE
feat: replace hardGoTo with navigate

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
reach-router's navigate does everything we need, preserves navigation
history and should be more accessible

### CODE REVIEW

File: client/src/client-only-routes/ShowSettings.js [hunk 1]
```diff
@@ -57,7 +57,6 @@ const mapStateToProps = createSelector(
 
 const mapDispatchToProps = {
   createFlashMessage,
-  navigate,
```
Comment (ShaunSHamilton) [inline client/src/client-only-routes/ShowSettings.js:60]: This is still in the propTypes... unless, you left it to be taken care of by one of the TS refactors

**Review by raisedadead — CHANGES_REQUESTED** [review]
Tests.

File: client/src/client-only-routes/ShowSettings.js (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/client-only-routes/ShowSettings.js]: Thanks, Shaun, that's fixed now.

**Comment by ojeytonwilliams** [comment]
This proved to be more complicated than I thought.  Might be worth revisiting when we rework the backend, but not until then.

**Pull request closed without merging**